### PR TITLE
feat: Convert paths inside `sources` set to relative to project root

### LINKED ISSUE #29: Prevent private file paths from being exposed to third parties
The current extraction results include private file paths.

In SSR/SSG setups like SvelteKit, the results extracted by the Vite plugin are statically embedded into both the server-side and client-side code, which is efficient. However, this unintentionally exposes private file paths to the client.

Example extraction result:

```js
// ...
        "type": {
          "kind": "union",
          "types": [
            { "kind": "undefined" },
            {
              "kind": "constructible",
              "name": '"/Users/*****/*****/github.com/ciscorn/my-project/node_modules/.pnpm/external-package/node_modules/external-package".LngLat',
              "constructors": [
                [
                  {
                    "name": "lng",
                    "isOptional": false,
                    "type": { "kind": "number" }
                  },
                  {
                    "name": "lat",
                    "isOptional": false,
                    "type": { "kind": "number" }
                  }
                ]
              ],
              "sources": /* @__PURE__ */ new Set([
                "/Users/*****/*****/github.com/ciscorn/my-project/node_modules/.pnpm/external-package/node_modules/external-package/dist/types.d.ts"
              ])
            },
// ...
```

### PR DESCRIPTION
**Pull request by xeho91**
Resolves #29

We could also remove the `<process-cwd>` from the snapshot serializer.
But... on the second thought. It could also help us detect when this feature isn't working?

### CODE REVIEW

File: packages/svelte-docgen/src/parser/type/constructible.test.ts [hunk 1]
```diff
@@ -530,11 +530,11 @@ describe("Constructible", () => {
 			              "alias": "Iterable",
 			              "kind": "interface",
 			              "members": Map {
-			                "__@iterator@88" => {
+			                "__@iterator@19" => {
```
Comment (xeho91) [inline packages/svelte-docgen/src/parser/type/constructible.test.ts:533]: . Unrelated to the subject of this issue, but:

No idea why those numbers keep changing. It has to do with internals inside TypeScript Compiler API.

Comment (xeho91) [reply packages/svelte-docgen/src/parser/type/constructible.test.ts]: **NOTES:**

The numbers change when running a test script from:

- workspace root
- package root

For the sake of CI - use the workspace root.

**Review by ciscorn — APPROVED** [review]
The hiding seems to be working as I expected!

That said, I’m curious to hear others’ thoughts on how this path masking should be.

**Comment by xeho91** [comment]
Alright, I'm merging this one to unblock the progress.

We haven't received any feedback, so any potential change need can always be implemented in the future.

**Pull request merged**